cms/bundles/views/preview.py

```diff
@@ -109,16 +133,104 @@ def get(self, request: "HttpRequest", *args: Any, **kwargs: Any) -> TemplateResp
             data=log_data_entry,
         )
 
-        if user_can_manage_bundles(self.request.user):
-            pages_in_bundle = bundle.get_bundled_pages(specific=True)
-        else:
-            pages_in_bundle = bundle.get_pages_for_previewers()
+        pages_in_bundle = self.get_pages_in_bundle(bundle)
 
         request.is_dummy = True  # type: ignore[attr-defined]
         request.is_preview = True  # type: ignore[attr-defined]
         request.preview_mode = "bundle-preview"  # type: ignore[attr-defined]
 
         context["bundle_inspect_url"] = reverse("bundle:inspect", args=[bundle_id])
-        context["preview_items"] = get_preview_items_for_bundle(bundle, release_calendar_page.id, pages_in_bundle)
+
+        # Fetch bundle contents to include datasets in preview items
+        bundle_contents = self.get_bundle_contents(bundle)
+
+        context["preview_items"] = get_preview_items_for_bundle(
+            bundle, release_calendar_page.id, pages_in_bundle, bundle_contents
+        )
 
         return TemplateResponse(request, release_calendar_page.get_template(request), context)
+
+
+class PreviewBundleDatasetView(BundleContentsMixin, TemplateView):
+    template_name = "templates/bundles/preview.html"
+    http_method_names: Sequence[str] = ["get"]
+
+    @staticmethod
+    def find_dataset_preview_url(
+        bundle_contents: dict[str, Any], dataset_id: str, edition_id: str, version_id: str
+    ) -> str | None:
+        """Find the preview URL for a specific dataset in the bundle contents.
+
+        Args:
+            bundle_contents: The bundle contents from the API
+            dataset_id: The dataset identifier
+            edition_id: The edition identifier
+            version_id: The version identifier
+
+        Returns:
+            The preview URL if found, None otherwise
+        """
+        for item in bundle_contents.get("items", []):
+            if item.get("content_type") != "DATASET":
+                continue
+
+            metadata = item.get("metadata", {})
+            if (
+                metadata.get("dataset_id") == dataset_id
+                and metadata.get("edition_id") == edition_id
+                and metadata.get("version_id") == version_id
+            ):
+                return item.get("links", {}).get("preview", "") or None
+
+        return None
+
+    def get(self, request: "HttpRequest", *args: Any, **kwargs: Any) -> "TemplateResponse | HttpResponseRedirect":
```

Comment:
Trying to understand this function, it looks like it can return a specific dataset preview and/or all the page previews? On thinking about it, it makes sense to me because of how the two different objects are previewed very differently, but maybe worth a doc string explaining?